docs/src/Guides/20 Extensions.md

```diff
@@ -1,248 +1,581 @@
+from interactions import Extension, slash_command, SlashContext
+
+class MyExtension(Extension):
+    @slash_command()
+    async def test(self, ctx: SlashContext):
+        await ctx.send("Hello world!")
 ```
-As you can see, there's one extra bit, a function called `setup`, this function acts as an entry point for interactions.py,
-so it knows how to load the extension properly.
 
-To load a extension, you simply add the following to your `main` script, just above `bot.start`:
+As you can see, they're almost identical to how you declare slash commands in your main bot file, even using the same decorator.
+The only difference is the `self` variable - this is the instance of the extension that the command is being called in, and is
+standard for functions inside of classes. Events follow a similar principal.
+
+interactions.py will automatically add all commands and events to the bot when you load the extension (discussed later),
+so you don't need to worry about that.
+
+#### Accessing the Bot
+
+You may have noticed that the `self` variable is being used in the example above. This is because the bot is passed to the extension
+when it's loaded and is stored in the `self.bot` variable. This means you can access the bot from anywhere in your extension,
 
 ```python
-...
+class MyExtension(Extension):
+    @slash_command()
+    async def test(self, ctx: SlashContext):
+        await ctx.send(f"Hello, I'm {self.bot.user.mention}!")
+```
 
-bot.load_extension("Filename_here")
+This also allows you to share data between extensions and the main bot itself. `Client` allows storing data in unused attributes,
+so you can do something like this:
 
+```python
+from interactions import Client
+
+# main.py
+bot = Client(...)
+bot.my_data = "Hello world"
+
+# extension.py
+class MyExtension(Extension):
+    @slash_command()
+    async def test(self, ctx: SlashContext):
+        await ctx.send(f"My data: {self.bot.my_data}!")
+```
+
+### Loading the Extension
+
+Now that you've got your extension, you need to load it.
+
+Let's pretend the extension is in a file called `extension.py`, and it looks like the command example:
+
+```python
+from interactions import Extension, slash_command, SlashContext
+
+class MyExtension(Extension):
+    @slash_command()
+    async def test(self, ctx: SlashContext):
+        await ctx.send("Hello world!")
+```
+
+Now, let's say you have a file called `main.py` in the same directory that actually has the bot in it:
+
+```python
+bot = Client(...)
 bot.start("token")
 ```
 
-Now, for the cool bit of Extensions, reloading. Extensions allow you to edit your code, and reload it, without restarting the bot.
-To do this, simply run `bot.reload_extension("Filename_here")` and your new code will be used. Bare in mind any tasks your extension
-is doing will be abruptly stopped.
+To load the extension, you just need to use `bot.load_extension("filename_in_import_style")` before `bot.start`. So, in this case, it would look like this:
```

Comment:
```suggestion
To load the extension, you just need to use `bot.load_extension("filename.in_import_style")` before `bot.start`. So, in this case, it would look like this:
```
Suggest adding a dot, to differentiate between snake_case and import